packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.ts

```diff
@@ -228,6 +238,95 @@ export default createESLintRule<Options, MessageIds>({
           data: {
             npmScope,
           },
+          fix(fixer) {
+            const targetImportProject = getTargetProjectBasedOnRelativeImport(
+              imp,
+              projectPath,
+              projectGraph,
+              sourceFilePath
+            );
+
+            if (targetImportProject) {
+              const indexTsPaths =
+                getBarrelEntryPointProjectNode(targetImportProject);
+
+              if (indexTsPaths && indexTsPaths.length > 0) {
+                const imports = ((node as any).specifiers || []).map(
+                  (specifier) => specifier.imported.name
+                );
+
+                // ignore non-named imports for now, like "import '../../some-other-lib/src'"
+                if (imports.length === 0) {
+                  return;
+                }
+
+                // process each potential entry point and try to find the imports
+                const importsToRemap = [];
+
+                for (const entryPointPath of indexTsPaths) {
+                  for (const importMember of imports) {
+                    const importPath = getRelativeImportPath(
+                      importMember,
+                      entryPointPath.path,
+                      sourceProject.data.sourceRoot
+                    );
+
+                    if (importPath) {
+                      importsToRemap.push({
+                        member: importMember,
+                        importScope: entryPointPath.importScope,
+                      });
+                    } else {
+                      // we cannot remap, so leave it as is
+                      importsToRemap.push({
```

Comment:
What would be the scenario when this happens? A missing path in tsconfig?

Can we have this scenario in tests?